base/commands/migration/start_stages_it_test.go

```diff
@@ -0,0 +1,133 @@
+//go:build std || migration
+
+package migration_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	_ "github.com/hazelcast/hazelcast-commandline-client/base"
+	_ "github.com/hazelcast/hazelcast-commandline-client/base/commands"
+	"github.com/hazelcast/hazelcast-commandline-client/base/commands/migration"
+	"github.com/hazelcast/hazelcast-commandline-client/clc/paths"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+	hz "github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMigrationStages(t *testing.T) {
+	testCases := []struct {
+		name                string
+		statusMapStateFiles []string
+		expectedErr         error
+	}{
+		{
+			name: "successful",
+			statusMapStateFiles: []string{
+				"testdata/start/migration_success_initial.json",
+				"testdata/start/migration_success_completed.json",
+			},
+		},
+		{
+			name: "failure",
+			statusMapStateFiles: []string{
+				"testdata/start/migration_success_initial.json",
+				"testdata/start/migration_success_failure.json",
+			},
+			expectedErr: errors.New("Failed migrating IMAP: imap5: some error"),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			startMigrationTest(t, tc.expectedErr, tc.statusMapStateFiles)
+		})
+	}
+}
+
+func startMigrationTest(t *testing.T, expectedErr error, statusMapStateFiles []string) {
+	tcx := it.TestContext{T: t}
+	ctx := context.Background()
+	tcx.Tester(func(tcx it.TestContext) {
+		ci := hz.NewClientInternal(tcx.Client)
+		createMapping(ctx, tcx)
+		createMemberLogs(t, ctx, ci)
+		defer removeMembersLogs(ctx, ci)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		var execErr error
+		go tcx.WithReset(func() {
+			defer wg.Done()
+			execErr = tcx.CLC().Execute(ctx, "start", "dmt-config", "--yes")
+		})
+		c := make(chan string, 1)
+		wg.Add(1)
+		go findMigrationID(ctx, tcx, c)
+		mID := <-c
+		wg.Done()
```

Comment:
I think waitGroup will not be required here if `c` is unbuffered.

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/398/commits/f50a1de3621a0e85a2f50804eca9223edc4fd15c